src/signal.cr

```diff
@@ -327,7 +327,21 @@ fun __crystal_sigfault_handler(sig : LibC::Int, addr : Void*)
   stack_top = Pointer(Void).new(Fiber.current.@stack.address - 4096)
 
   if stack_top <= addr < Fiber.current.@stack_bottom
-    LibC.dprintf 2, "Stack overflow (e.g., infinite or very deep recursion)\n"
+    LibC.dprintf 2,
```

Comment:
An error message should be concise, not explain the general concept of stack overflow errors. This is fine as is.

Reply:
Please let's not detail the concept of stack overflow in a runtime error message. While it is likely a code issue, this error message is presented to the *user*, not the *developer* and should not try to explain a developer's tasks to a user.

Reply:
Maybe we could consider a more detailed error message for development builds, but there is no clear indicator for that. Maybe `flag?(:release)` would do, but that's not its purpose.

Reply:
Ok you are right with that point. The user will see this error with code for the language the developer is using and that comes a bit unprofessional too. Like the user could come "yeah, developer I think you have to do that and this to fix this error in my application".
Only compile time errors should be as detailed as this error.